addons/binding/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/handler/HiveThermostatHandler.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2010-2019 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.hive.internal.handler;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.hive.internal.HiveBindingConstants;
+import org.openhab.binding.hive.internal.classes.HiveAttributes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Chris Foot - Initial contribution
+ */
+
+public class HiveThermostatHandler extends BaseThingHandler {
```

Comment:
Please annotate all your classes (except dto/model) with `@NonNullByDefault`.
The compiler will force you to assign values or annotate fields with `@Nullable`. -> Helps to prevent NPEs (Null pointer exceptions)